plugins/datastreamer_plugin/ui/topic_selection_dialog/Configuration.cpp

```diff
@@ -0,0 +1,106 @@
+//
+// eProsima Fast DDS Monitor is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with eProsima Fast DDS Monitor. If not, see <https://www.gnu.org/licenses/>.
+
+/**
+ * @file Configuration.cpp
+ */
+
+#include "Configuration.hpp"
+#include "utils/utils.hpp"
+
+namespace eprosima {
+namespace plotjuggler {
+namespace ui {
+
+Configuration::Configuration()
+{
+}
+
+Configuration::Configuration(const QString& prefix)
+{
+    load_default_settings(prefix);
+}
+
+////////////////////
+// Save Load configuration
+bool Configuration::xmlSaveState(
+        QDomDocument& doc,
+        QDomElement& parent_element) const
+{
+    DEBUG("Saving XML configuration state");
+
+    QDomElement max_elem = doc.createElement(MAX_ARRAY_SIZE_SETTINGS_TAG);
+    max_elem.setAttribute("value", QString::number(data_type_configuration.max_array_size));
+    parent_element.appendChild(max_elem);
+
+    QDomElement stamp_elem = doc.createElement(USE_HEADER_STAMP_SETTINGS_TAG);
+    stamp_elem.setAttribute("value", data_type_configuration.use_header_stamp ? "true" : "false");
+    parent_element.appendChild(stamp_elem);
+
+    QDomElement discard_elem = doc.createElement(DISCARD_LARGE_ARRAYS_SETTINGS_TAG);
+    discard_elem.setAttribute("value", data_type_configuration.discard_large_arrays ? "true" : "false");
+    parent_element.appendChild(discard_elem);
+
+    return true;
+}
+
+bool Configuration::xmlLoadState(
+        const QDomElement& parent_element)
+{
+    DEBUG("Loading XML configuration state");
+
+    QDomElement max_elem = parent_element.firstChildElement(MAX_ARRAY_SIZE_SETTINGS_TAG);
+    data_type_configuration.max_array_size = max_elem.attribute("value").toInt();
+
+    QDomElement stamp_elem = parent_element.firstChildElement(USE_HEADER_STAMP_SETTINGS_TAG);
+    data_type_configuration.use_header_stamp = (stamp_elem.attribute("value") == "true");
+
+    QDomElement discard_elem = parent_element.firstChildElement(DISCARD_LARGE_ARRAYS_SETTINGS_TAG);
+    data_type_configuration.discard_large_arrays = (discard_elem.attribute("value") == "true");
+
+    return true;
+}
+
+void Configuration::save_default_settings(
+        const QString& prefix) const
+{
+    // Get default Qt settings
+    QSettings settings;
+
+    settings.setValue(prefix + MAX_ARRAY_SIZE_SETTINGS_TAG, data_type_configuration.max_array_size);
+    settings.setValue(prefix + USE_HEADER_STAMP_SETTINGS_TAG, data_type_configuration.use_header_stamp);
+    settings.setValue(prefix + DISCARD_LARGE_ARRAYS_SETTINGS_TAG, data_type_configuration.discard_large_arrays);
+    settings.setValue(prefix + XML_DATATYPE_FILES_SETTINGS_TAG, xml_datatypes_files);
+    settings.setValue(prefix + DOMAIN_ID_SETTINGS_TAG, domain_id);
+}
```

Comment:
A new `QSettings` object is being built and destroyed at the end of the scope, what for?